audio: Replace inferred array lengths with `BUFFER_SIZE`

atomkernel0 opened the pull request:
### Summary
Fix stable Rust builds (E0658) by replacing `_`-inferred array lengths with explicit `BUFFER_SIZE` in `crates/audio/src/audio.rs`. No behavior changes.

### Context
Stable Rust doesn’t allow `_` as an array length in type position (E0658; see rust-lang/rust#85077), causing two compile errors in `audio.rs`.

I got these two compilation errors : 

```
error[E0658]: using `_` for array lengths is unstable
   --> crates/audio/src/audio.rs:131:40
    |
131 |                     let mut buf: [i16; _] = buffer.map(|s| s.to_sample());
    |                                        ^
    |
    = note: see issue #85077 <https://github.com/rust-lang/rust/issues/85077> for more information

error[E0658]: using `_` for array lengths is unstable
   --> crates/audio/src/audio.rs:172:43
    |
172 |                 let mut int_buffer: [i16; _] = buffer.map(|s| s.to_sample());
    |                                           ^
    |
    = note: see issue #85077 <https://github.com/rust-lang/rust/issues/85077> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `audio` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
```

`BUFFER_SIZE` matches the surrounding const generics:
- `inspect_buffer::<BUFFER_SIZE, _>(...)`
- `process_buffer::<BUFFER_SIZE, _>(...)`

### Testing
Built successfully on stable:
- rustc 1.88.0 (6b00bc388 2025-06-23), LLVM 19.1.4
- target: x86_64-unknown-linux-gnu (Void Linux)

### Impact
Build-only fix; no functional changes.

### Release notes
Fix: Restore stable Rust build by replacing `_` array lengths with `BUFFER_SIZE` in `crates/audio/src/audio.rs` (E0658).

atomkernel0 commented:
@cla-bot check

maxdeviant commented:
We already build on Rust stable, so this seems like something specific to your environment.

> Built successfully on stable:
> 
> * rustc 1.88.0 (6b00bc388 2025-06-23), LLVM 19.1.4

I see you're building with Rust 1.88, but we have Rust 1.89 in our `rust-toolchain.toml`: https://github.com/zed-industries/zed/blob/ffa23d25e315eda8079664fb8fe2d8618003ef4b/rust-toolchain.toml#L2

What happens if you build on Rust 1.89?

atomkernel0 committed:
fix(audio): replace unstable '_' array length with BUFFER_SIZE for stable Rust

On stable Rust, using \`_\` to infer array lengths in type position is not stabilized (E0658). This change replaces the inferred lengths with explicit BUFFER_SIZE in audio.rs, matching the const generic used by the buffer processing. No functional changes; fixes build on stable. Refs: rust-lang/rust#85077.
